app/models/course/enrol_request.rb

```diff
@@ -1,5 +1,17 @@
 # frozen_string_literal: true
 class Course::EnrolRequest < ActiveRecord::Base
+  validate :validate_email_not_invited, on: :create
+
   belongs_to :course, inverse_of: :enrol_requests
   belongs_to :user, inverse_of: :course_enrol_requests
+
+  private
+
+  # Ensure that there are no enrol requests by the same users that are invited.
+  def validate_email_not_invited
+    email = user.emails.map(&:email) & course.invitations.map(&:email)
+    return if email.blank?
+
+    errors.add(:base, :invitation_email_exist)
```

Comment:
@weiqingtoh Does this error get correctly displayed in the page  ?  I think it might not be displayed cause we are using an ActiveModel to wrap the object in the registration page.